src/videojs-hls.js

```diff
@@ -321,12 +321,21 @@ videojs.Hls.prototype.play = function() {
     this.mediaIndex = 0;
   }
 
-  // seek to the latest safe point in the media timeline when first
-  // playing live streams
-  if (this.duration() === Infinity &&
-      this.playlists.media() &&
-      !this.player().hasClass('vjs-has-started')) {
-    this.setCurrentTime(this.seekable().end(0));
+  // we may need to seek to begin playing safely for live playlists
+  if (this.duration() === Infinity) {
+
+    // if this is the first time we're playing the stream or we're
+    // ahead of the latest safe playback position, seek to the live
+    // point
+    if (!this.player().hasClass('vjs-has-started') ||
+        this.currentTime() > this.seekable().end(0)) {
+      this.setCurrentTime(this.seekable().end(0));
+
+    } else if (this.currentTime() < this.seekable().start(0)) {
+      // if the viewer has paused and we fell out of the live window,
+      // seek forward to the earliest available position
```

Comment:
in this case, do we want to seek to the earliest position or to the "3 segments behind live"/"live point" position?